### DISCUSSION TITLE
Disk Harvester harvests too many snaps

### DISCUSSION THREAD

**Discussion by efwe** [ClementTsang/bottom#778] [Q&A] [answered]
hullo,

I use bottom on Ubuntu 22.04 and the default behavior of the disk widget shows tons of mount-points  of type `squashfs` used by `snap`. For me this information is not so helpful and I'm more interested in the status of my _physical_ disks.

I could try to improve the situation but I would need some pointers how to implement such a 'filtering' feature.
I also browsed through the heim.rs code and found `is_physical` in  [filesystem.rs](https://github.com/heim-rs/heim/blob/b292f1535bb27c03800cdb7509fa81a40859fbbb/heim-disk/src/filesystem.rs) - which could be something for the harvesting.

in any way - what do you guys see in the `disks` section and are you happy about it in the default config?

keep up the good work!
~fw

**Comment by ClementTsang** [comment, marked as answer]
There already [is a basic filtering feature](https://clementtsang.github.io/bottom/0.6.8/configuration/config-file/data-filtering/) for disks and mounts. I guess you could use a regex filter on the mounts and ignore any entry with `snap`.

I might look at adding a general "ignore snaps" or "only physical" option though as that seems to be a popular use case.

**Reply by efwe** [reply]
sorry that I missed that feature. It worked of course instantly - sorry for the RTFM question.

**Reply by ClementTsang** [reply]
All good, helps highlight that feature discoverability is a bit of a weak spot right now. I should really put this info in an easy spot within the application.
